functions/index.js

```diff
@@ -1,7 +1,16 @@
 const { onRequest } = require("firebase-functions/v2/https");
 const logger = require("firebase-functions/logger");
 
-exports.helloWorld = onRequest((request, response) => {
-  logger.info("Hello logs!", { structuredData: true });
-  response.send("Hello from Firebase!");
+exports.getWeatherInformation = onRequest(async (request, response) => {
+  const prefectureEn = request.query.prefecture;
+  const api = process.env.WEATHER_API_KEY;
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${prefectureEn},JP&appid=${api}&units=metric&lang=ja`;
+  const data = await fetch(url);
+  const json = await data.json();
+  logger.info("天気情報取得", { structuredData: true });
+  response.set("Access-Control-Allow-Headers", "*");
```

Comment:
ここの`Access-Control-Allow-Headers`の設定ですが"*"に設定してしまうと自分のサービス以外からもAPIにアクセスできてしまいAPIキーを悪用されてしまう可能性があるのでアプリケーションのドメイン（`getweatherinformation-afq4w33w3q-uc.a.run.app`など）に設定しておく必要があるかと思います。

![スクリーンショット 2024-07-01 17 30 21](https://github.com/msuzuna/improve-quality/assets/57705206/201bebdb-2a47-480f-bf4a-7bb95df5626a)

Reply:
ちょっとこのまま動くかわからないんですけど、プロダクションコードの時のみ制限を掛ける場合は下記のようなコードになるかと

```suggestion
  response.set(
    "Access-Control-Allow-Headers",
    process.env.NODE_ENV === 'production' 
      ? "getweatherinformation-afq4w33w3q-uc.a.run.app"
      : "*"
  );
```

Reply:
修正対応中です><

Reply:
すみません、色々試したのですがなかなかうまくいきませんでした><

ひとまず、`Access-Control-Allow-Headers`は利用できるヘッダを指定する、とのことだったので`"Origin, Methods"`を指定しました！
a4fd4513315f4d37ddaee624540ff65a8c1b2f0d

ドメインを制限するのは`Access-Control-Allow-Origin`の方みたいなのですが、firebase functiionをローカルから都度デプロイしている都合上、環境に合わせて変える、ということが自分が試した範囲だとできなさそうでした><
実装対応中は現状ママにしておき、本PRをマージできたら別PRにてドメインを制限しようかと思うのですが、いかがでしょうか？
https://github.com/msuzuna/improve-quality/pull/13

Reply:
あ、これ`Access-Control-Allow-Origin`が正しいですね…普通に間違ってました。

> firebase functiionをローカルから都度デプロイしている都合上、環境に合わせて変える、ということが自分が試した範囲だとできなさそうでした><

なるほど！一応手元でビルドする時も本番ビルドと開発ビルドは分けられると思うのでバンドラーの設定回りいじればできる気がしますがとりあえずは別途対応でもいいかもしれません。

Reply:
@ken7253 
ご確認ありがとうございます！今回チャレンジしたい本筋とはズレる部分なので、一旦今回はこのまま進めようと思います！